### PR TITLE
Add Jetpack version based detection

### PR DESCRIPTION
**Pull request by ybadaoui-ostorlab**
This implementation introduces a version-based detection mechanism for the Jetpack plugin. The code extracts the plugin version from its `readme.txt` file and compares it against a predefined list of vulnerable version ranges. If the extracted version falls within any of the specified ranges, the plugin is flagged as vulnerable. This approach ensures accurate identification of affected versions by directly parsing and evaluating the version string.

Reference: [Jetpack 13.9.1 Security Update](https://jetpack.com/blog/jetpack-13-9-1-critical-security-update/).

### CODE REVIEW

File: agent/exploits/jetpack_version_detection.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        target_endpoint = urlparse.urljoin(target.origin, self.check_request.path)
+
+        try:
+            req = requests.Request(
+                method=self.check_request.method,
+                url=target_endpoint,
+                data=self.check_request.data,
+            ).prepare()
+            resp = session.send(req, timeout=DEFAULT_TIMEOUT.seconds)
+        except requests_exceptions.RequestException:
```
Comment (nmasdoufi-ol) [inline agent/exploits/jetpack_version_detection.py:172]: Add logging for network errors (in the `except` block) to make debugging easier.

File: agent/exploits/jetpack_version_detection.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+import requests
+from requests import exceptions as requests_exceptions
+from semver import Version
+
+from agent import definitions
+from agent import exploits_registry
+from agent.exploits import webexploit
+
+MAX_REDIRECTS = 2
+DEFAULT_TIMEOUT = datetime.timedelta(seconds=90)
```
Comment (benyissa) [inline agent/exploits/jetpack_version_detection.py:14]: Isn't 90 seconds too much?

**Comment by 3asm** [comment]
Please fix the title and improve the description.

File: agent/exploits/jetpack_version_detection.py (hunk 1 above)
Comment (ybadaoui-ostorlab) [reply agent/exploits/jetpack_version_detection.py:172]: done!

File: agent/exploits/jetpack_version_detection.py (hunk 2 above)
Comment (ybadaoui-ostorlab) [reply agent/exploits/jetpack_version_detection.py:14]: that's what is used in older exploits

File: agent/exploits/jetpack_version_detection.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+        vulnerabilities: list[definitions.Vulnerability] = []
+
+        target_endpoint = urlparse.urljoin(target.origin, self.check_request.path)
+
+        try:
+            req = requests.Request(
+                method=self.check_request.method,
+                url=target_endpoint,
+                data=self.check_request.data,
```
Comment (ohachimOs) [inline agent/exploits/jetpack_version_detection.py:169]: Do we need to raise for status? or is that not a case that can happen here?

File: agent/exploits/jetpack_version_detection.py [hunk 4]
```diff
@@ -0,0 +1,202 @@
+import datetime
+import re
+from urllib import parse as urlparse
+
+import requests
+from requests import exceptions as requests_exceptions
+from semver import Version
```
[the remark this answers is not in the document]
Comment (mohsinenar) [reply agent/exploits/jetpack_version_detection.py:7]: @rabsondev Did you do yours??

[the remark this answers is not in the document]
Comment (rabsondev) [reply agent/exploits/jetpack_version_detection.py:7]: > @rabsondev Did you do yours??

499 done, 1 more to go.

File: agent/exploits/jetpack_version_detection.py [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+            ).prepare()
+            resp = session.send(req, timeout=DEFAULT_TIMEOUT.seconds)
+        except requests_exceptions.RequestException:
+            return vulnerabilities
+
+        if (matched := self.version_pattern.findall(resp.text)) != []:
+            for extracted_version in matched:
+                if _is_vulnerable(extracted_version) is True:
+                    vulnerability = self._create_vulnerability(target)
+                    vulnerabilities.append(vulnerability)
```
Comment (mohsinenar) [inline agent/exploits/jetpack_version_detection.py:179]: It not possible to have multiple version and report multiple vulnz for a single target! 
This hould not be a loop

File: agent/exploits/jetpack_version_detection.py (hunk 3 above)
Comment (ybadaoui-ostorlab) [reply agent/exploits/jetpack_version_detection.py:169]: no

File: agent/exploits/jetpack_version_detection.py (hunk 5 above)
Comment (ybadaoui-ostorlab) [reply agent/exploits/jetpack_version_detection.py:179]: just forget to change it. now it's fixed! thanks

File: agent/exploits/jetpack_version_detection.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ybadaoui-ostorlab) [reply agent/exploits/jetpack_version_detection.py:7]: fixed!

File: agent/exploits/jetpack_version_detection.py [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+    ("4.8.0", "4.8.4"),
+    ("4.7.0", "4.7.3"),
+    ("4.6.0", "4.6.2"),
+    ("4.5.0", "4.5.2"),
+    ("4.4.0", "4.4.4"),
+    ("4.3.0", "4.3.4"),
+    ("4.2.0", "4.2.4"),
+    ("4.1.0", "4.1.3"),
+    ("4.0.0", "4.0.6"),
+    (None, "3.9.9"),
```
Comment (oussamaessaji) [inline agent/exploits/jetpack_version_detection.py:128]: why None, `0.0.0` won't work???

**Review by 3asm — APPROVED** [review]

**Pull request merged**